src/master-playlist-controller.js

```diff
@@ -429,8 +402,17 @@ export class MasterPlaylistController extends videojs.EventTarget {
       let updatedPlaylist = this.masterPlaylistLoader_.media();
 
       if (!updatedPlaylist) {
-        // select the initial variant
-        this.initialMedia_ = this.selectPlaylist();
+        let selectedMedia;
```

Comment:
Looked into this a bit more, its because if you set `this.initialMedia_` and then try and load that media, but the request fails e.g. 404, when on the final rendition, we get back to here, and `updatedPlaylist` is still `undefined` because we haven't successfully loaded one yet, but `initialMedia_` has been set from the previous time here. I think it would make sense to set `initialMedia_` in the `loadedmetadata` handler if it hasnt been set yet as the first time we get into `loadedmetadata` is after the first media playlist has been loaded.